src/sentry/utils/retries.py

```diff
@@ -66,27 +69,38 @@ def delay(i):
         self.delay = delay
         self.exceptions = exceptions
         self.clock = time
+        self.metric_instance = metric_instance
+        self.metric_tags = metric_tags or {}
 
     def __call__(self, function):
         start = self.clock.time()
-        for i in itertools.count(1):
-            try:
-                return function()
-            except self.exceptions as error:
-                delay = self.delay(i)
-                now = self.clock.time()
-                if (now + delay) > (start + self.timeout):
-                    raise RetryException(
-                        'Could not successfully execute %r within %.3f seconds (%s attempts.)' %
-                        (function, now - start, i),
-                        error,
-                    )
-                else:
-                    logger.debug(
-                        'Failed to execute %r due to %r on attempt #%s, retrying in %s seconds...',
-                        function,
-                        error,
-                        i,
-                        delay,
-                    )
-                    self.clock.sleep(delay)
+        try:
+            for i in itertools.count(1):
+                try:
+                    return function()
+                except self.exceptions as error:
+                    delay = self.delay(i)
+                    now = self.clock.time()
+                    if (now + delay) > (start + self.timeout):
+                        raise RetryException(
+                            'Could not successfully execute %r within %.3f seconds (%s attempts.)' %
+                            (function, now - start, i),
+                            error,
+                        )
+                    else:
+                        logger.debug(
+                            'Failed to execute %r due to %r on attempt #%s, retrying in %s seconds...',
+                            function,
+                            error,
+                            i,
+                            delay,
+                        )
+                        self.clock.sleep(delay)
+        finally:
+            if self.metric_instance:
+                metrics.timing(
+                    'timedretrypolicy.duration',
```

Comment:
Can you change this name to be more specific that this is for acquisition and not held time?

Reply:
Well. This technically has nothing to do with locks. So it could in theory be anything, but the metric name is `timedretrypolicy` not `lock` for that reason. :)

Reply:
Dang

Reply:
I guess to be more explicitly, I'd be putting `lock.acquire:foo` or something into the instance name.

Reply:
That sounds good to me, keeping the abstraction separate here is a good point